src/components/ContactCard/ContactForm.jsx

```diff
@@ -136,24 +135,34 @@ class ContactForm extends React.Component {
     const { onCancel, t } = this.props;
     return (
       <Form
+        mutators={{ ...arrayMutators }}
         onSubmit={this.formDataToContact}
+        initialValues={initialFieldValues}
         render={({ handleSubmit }) => (
           <div>
             <form onSubmit={handleSubmit} className="contact-form">
               <div className="contact-form__fields">
-                {fields.map(({ name, icon, type, required, hasLabel }) => (
-                  <ContactFormField
-                    key={name}
-                    icon={icon}
-                    name={name}
-                    label={t(`field.${name}`)}
-                    placeholder={t(`placeholder.${name}`)}
-                    type={type}
-                    inputWithLabel={hasLabel}
-                    required={required}
-                    labelPlaceholder={t("placeholder.label")}
-                  />
-                ))}
+                {fields.map(
+                  ({ name, icon, type, required, hasLabel, isArray }) => (
+                    <ContactFormField
+                      key={name}
+                      name={name}
+                      icon={icon}
+                      label={t(`field.${name}`)}
+                      isArray={isArray}
+                      renderInput={inputName => (
+                        <ContactFieldInput
+                          name={inputName}
+                          type={type}
+                          placeholder={t(`placeholder.${name}`)}
+                          required={required}
+                          withLabel={hasLabel}
+                          labelPlaceholder={t("placeholder.label")}
+                        />
```

Comment:
I've changed this, I think it's better but still not pefect.

The structure is always the same

```
< ContactFormField>
  < ContactFieldInput />
</ContactFormField>
```

But `ContactFieldInput` needs a lot of props that are unrelated to `ContactFormField`.

Before, we passed a lot of props to `ContactFormField`, who then used a few, and passed the rest down to `ContactFieldInput`. By using a renderProp, we can give each component only the props it really needs to do it's job.

What I still don't like too much is that we could render *anything* in `ContactFormField.renderInput`, when in reality, we really expect a `ContactFieldInput`, or at least a `Field` from final-form.

Reply:
The new way seems ok to me. The renderInput being restricted is not a real problem for me. A line in the documentation should do the trick, and the code will fail if ever someone tries to do something else.